geo/src/algorithm/sweep/point.rs

```diff
@@ -29,17 +29,17 @@ impl<T: GeoNum> std::fmt::Debug for SweepPoint<T> {
 /// coordinate.
 impl<T: GeoNum> PartialOrd for SweepPoint<T> {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        match self.0.x.partial_cmp(&other.0.x) {
-            Some(Ordering::Equal) => self.0.y.partial_cmp(&other.0.y),
-            o => o,
-        }
+        Some(self.cmp(other))
     }
 }
 
 /// Derive `Ord` from `PartialOrd` and expect to not fail.
 impl<T: GeoNum> Ord for SweepPoint<T> {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.partial_cmp(other).unwrap()
+        match self.0.x.total_cmp(&other.0.x) {
```

Comment:
Should we implement `TotalOrd` on some tuples, at least, to make this look nicer? The users might expect it to work similar to `Ord`

Reply:
This reasonable piece of feedback lead me down a rabbit hole. It might have been an over reaction on my part...

Ultimately, the reason for these changes are that I want to remove an `unwrap` from the code for the geometries we support. I introduced TotalOrd to do that. I don't actually care to support a general purpose TotalOrd trait. I realized:

1. GeoNum requires HasKernel, so arguably we only need to implement this functionality for the same types that we've defined HasKernel for (f32, f64, i16, i32, i64, isize).

2. Rather than try to keep in sync the implementations for HasKernel and GeoNum, since there's so little you can do with HasKernel without GeoNum, I though it made sense to simplify and and merge HasKernel's functionality into GeoNum.

I think this approach makes sense, but I'm prepared to back it out if it's controversial. @rmanoka in particular might have thoughts about that since the `HasKernel` trait was his devising (IIRC).

Reply:
Yeah, I guess implementing it for tuples isn't that easy. I'm fine with not having that, at least until someone complains.

https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.then_with can make this look nicer, but it doesn't really matter.